src/averageAges.js

```diff
@@ -55,9 +76,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => {
+    const boyHasMother = people.some(mother => mother.name === child.mother);
+    const isMale = child.sex === 'm';
+
+    return (onlyWithSon)
+      ? boyHasMother && isMale
+      : boyHasMother;
+  });
+
+  const differenceAge = children.map(child => (
+    child.born - people.find(mom => mom.name === child.mother).born
+  ));
+  const lastValue = averageAge(differenceAge, differenceAge.length);
+
+  return lastValue;
 }
 
+const averageAge = (array, length) => (
```

Comment:
Why do you need to pass length as an argument? You could calculate it in the function body